base/sparse/higherorderfns.jl

```diff
@@ -64,7 +64,8 @@ end
 
 
 # (2) map[!] entry points
-map{Tf}(f::Tf, A::SparseVecOrMat) = _noshapecheck_map(f, A)
+map{Tf}(f::Tf, A::SparseVector) = _noshapecheck_map(f, A)
+map{Tf}(f::Tf, A::SparseMatrixCSC) = _noshapecheck_map(f, A)
```

Comment:
Yes this change makes ambiguity detection pickier again.

This definition of `map` was less specific in the type of `A`, but more specific in argument count compared to the next one.